### PR TITLE
Reducing the number of IAM API calls made during enable cross-account…

### PR DESCRIPTION
**Pull request by StewartW**
… access

*Issue #, if available:* N/A

*Description of changes:*

I've changed the logic ever so slightly so that when enabling cross account access, it reduces the number of API calls  in the deployment account. Currently it will update the IAM policies per region (3 x Number of regions), meaning that it increases the chances of being throttled. 
This changes it to 3 API calls per account (or how many role/policy mappings are passed in). 

I've tried to make the change backwards compatible, just incase customers have downstream modifications to this lambda

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/lambda_codebase/enable_cross_account_access.py [hunk 1]
```diff
@@ -74,4 +77,7 @@ def lambda_handler(event, _):
             LOGGER.debug("%s could not be assumed (%s), continuing", account_id, err, exc_info=True)
             continue
 
+    update_iam(boto3, s3_buckets, kms_key_arns, role_policies)
+
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/lambda_codebase/enable_cross_account_access.py:81]: Suggesting to remove one of the two empty lines:
```suggestion
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/tests/test_iam.py [hunk 2]
```diff
@@ -44,3 +44,25 @@ def test_update_iam_cfn(cls):
     for policy in cls.policy.get('Statement'):
         if policy["Sid"] == "KMS":
             assert 'kms::12345678910::some_arn' in policy["Resource"]
+
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/tests/test_iam.py:47]: Requires another empty line before the new function:
```suggestion


```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/tests/test_iam.py [hunk 3]
```diff
@@ -44,3 +44,25 @@ def test_update_iam_cfn(cls):
...
+    for policy in cls.policy.get('Statement'):
+        if policy["Sid"] == "KMS":
+            assert 'kms::12345678910::kms_key_1' in policy["Resource"]
+            assert 'kms::12345678910::kms_key_2' in policy["Resource"]
+        if policy["Sid"] == "S3":
+            assert 'arn:aws:s3:::bucket2' in policy["Resource"]
+            assert 'arn:aws:s3:::bucket2/*' in policy["Resource"]
+            assert 'arn:aws:s3:::bucket1' in policy["Resource"]
+            assert 'arn:aws:s3:::bucket1/*' in policy["Resource"]
+
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/tests/test_iam.py:59]:
```suggestion


```

**Review by sbkok — CHANGES_REQUESTED** [review]
Minor suggestions only. Thanks for contributing this improvement!

**Review by sbkok — APPROVED** [review]
LGTM, thanks!

**Review by krisnielander — APPROVED** [review]
lgtm

**Pull request merged**